PhysicsTools/Heppy/python/analyzers/objects/JetAnalyzer.py

```diff
@@ -238,10 +239,12 @@ def process(self, event):
             else:
                 photons = [ g for g in event.selectedPhotons ] 
 
-        self.gamma_cleanJetsAll = cleanNearestJetOnly(self.cleanJetsAll, photons, self.jetGammaDR)
+#        self.gamma_cleanJetsAll = cleanNearestJetOnly(self.cleanJetsAll, photons, self.jetGammaDR)
+        self.gamma_cleanJetsAll = cleanNearestJetOnly(jetsEtaCut, photons+leptons, self.jetGammaDR)
         self.gamma_cleanJets    = [j for j in self.gamma_cleanJetsAll if abs(j.eta()) <  self.cfg_ana.jetEtaCentral ]
         self.gamma_cleanJetsFwd = [j for j in self.gamma_cleanJetsAll if abs(j.eta()) >= self.cfg_ana.jetEtaCentral ]
-        self.gamma_noIdCleanJetsAll = cleanNearestJetOnly(self.noIdCleanJetsAll, photons, self.jetGammaDR)
+#        self.gamma_noIdCleanJetsAll = cleanNearestJetOnly(self.noIdCleanJetsAll, photons, self.jetGammaDR)
+        self.gamma_noIdCleanJetsAll = cleanNearestJetOnly(self.jetsAllNoID, photons+leptons, self.jetGammaDR)
```

Comment:
Yes, this was actually done on purpose, since we want to clean jets from leptons and photons together (as a single collection equal to the sum of the two), with the same dR (=0.4). 
However, I agree this might be ambiguous.
I have just made a new commit, where I'm still doing the same but defining a different dR variable, called jetGammaLepDR. This way, the purpose of the cleaning should be non-ambiguous. 
Also, the jet/lepton cross-cleaning is unchanged, and it's done using jetLepDR, as usual. So, this change should not affect anyone else (I think MT2 is the only one using the gamma_jet collection?).